src/output-json-anomaly.c

```diff
@@ -134,6 +139,129 @@ static int AnomalyJson(ThreadVars *tv, JsonAnomalyLogThread *aft, const Packet *
     return TM_ECODE_OK;
 }
 
+static int AnomalyAppLayerDecoderEventJson(JsonAnomalyLogThread *aft,
+                        const Packet *p, AppLayerDecoderEvents *decoder_events,
+                        bool is_applayer, const char *layer, uint64_t tx_id)
+{
+    const char *alprotoname;
+    alprotoname = AppLayerGetProtoName(p->flow->alproto);
+
+    for (int i = 0; i < decoder_events->cnt; i++) {
+        MemBufferReset(aft->json_buffer);
+
+        json_t *js = CreateJSONHeader(p, LOG_DIR_PACKET, ANOMALY_EVENT_TYPE);
+        if (unlikely(js == NULL)) {
+            return TM_ECODE_OK;
+        }
+
+        json_t *ajs = json_object();
+        if (unlikely(ajs == NULL)) {
+            json_decref(js);
+            return TM_ECODE_OK;
+        }
+
+        JsonFiveTuple((const Packet *)p, LOG_DIR_PACKET, js);
```

Comment:
this is already called by CreateJSONHeader, is it needed here?

Reply:
Good catch.